control/vehicle_cmd_gate/src/vehicle_cmd_gate.cpp

```diff
@@ -381,69 +420,62 @@ void VehicleCmdGate::publishControlCommands(const Commands & commands)
 
   // Check engage
   if (!is_engaged_ || !start_request_->isAccepted()) {
-    filtered_commands.control.control = createStopControlCmd();
+    filtered_commands.control = createStopControlCmd();
   }
 
   // Check stopped after applying all gates
   start_request_->checkStopped(filtered_commands.control);
 
   // Apply limit filtering
-  filtered_commands.control.control = filterControlCommand(filtered_commands.control.control);
+  filtered_commands.control = filterControlCommand(filtered_commands.control);
 
-  // tmp: Create VehicleCmd
-  autoware_vehicle_msgs::msg::VehicleCommand vehicle_cmd;
-  vehicle_cmd.header = filtered_commands.control.header;
-  vehicle_cmd.control = filtered_commands.control.control;
-  vehicle_cmd.shift = filtered_commands.shift.shift;
-  vehicle_cmd.emergency = (use_emergency_handling_ && is_system_emergency_);
-
-  // Add frame_id to prevent RViz warnings
-  fillFrameId(&vehicle_cmd.header.frame_id, "base_link");
-  fillFrameId(&filtered_commands.control.header.frame_id, "base_link");
+  // tmp: Publish vehicle emergency status
+  autoware_auto_system_msgs::msg::EmergencyState vehicle_cmd_emergency;
+  vehicle_cmd_emergency.state = (use_emergency_handling_ && is_system_emergency_);
+  vehicle_cmd_emergency.stamp = filtered_commands.control.stamp;
 
   // Publish commands
-  vehicle_cmd_pub_->publish(vehicle_cmd);
+  vehicle_cmd_emergency_pub_->publish(vehicle_cmd_emergency);
   control_cmd_pub_->publish(filtered_commands.control);
 
   // Save ControlCmd to steering angle when disengaged
-  prev_control_cmd_ = filtered_commands.control.control;
+  prev_control_cmd_ = filtered_commands.control;
 }
 
 void VehicleCmdGate::publishEmergencyStopControlCommands()
 {
   const auto stamp = this->now();
 
   // ControlCommand
-  autoware_control_msgs::msg::ControlCommandStamped control_cmd;
-  control_cmd.header.stamp = stamp;
-  control_cmd.header.frame_id = "base_link";
-  control_cmd.control = createEmergencyStopControlCmd();
+  autoware_auto_control_msgs::msg::AckermannControlCommand control_cmd;
+  control_cmd.stamp = stamp;
+  control_cmd = createEmergencyStopControlCmd();
 
   // Check stopped after applying all gates
   start_request_->checkStopped(control_cmd);
 
-  // Shift
-  autoware_vehicle_msgs::msg::ShiftStamped shift;
-  shift.header.stamp = stamp;
-  shift.header.frame_id = "base_link";
-  shift.shift.data = autoware_vehicle_msgs::msg::Shift::NONE;
+  // gear
+  autoware_auto_vehicle_msgs::msg::GearCommand gear;
+  gear.stamp = stamp;
+  // default value is 0
```

Comment:
What do we assign to gear command here? When emergency, the ego vehicle tries to deccel and stop, so DRIVE?

Reply:
NONE is assigned as a default, so i didn't assign anything. 
If DRIVE should be assigned, then autoware_vehicle_msgs::msg::Shift::DRIVE was assigned.

Reply:
NONE(0) seems better, okay!